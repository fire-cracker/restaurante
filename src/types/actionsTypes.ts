export interface Action<T> {
  type: ActionType;
  payload?: T;
}

export enum ActionType {
  SIGNUP_REQUEST_PENDING = 'SIGNUP_REQUEST_PENDING',
  SIGNUP_REQUEST_SUCCESS = 'SIGNUP_REQUEST_SUCCESS',
  SIGNUP_REQUEST_FAILED = 'SIGNUP_REQUEST_FAILED',
  
  LOGIN_REQUEST_PENDING = 'LOGIN_REQUEST_PENDING',
  LOGIN_REQUEST_SUCCESS = 'LOGIN_REQUEST_SUCCESS',
  LOGIN_REQUEST_FAILED = 'LOGIN_REQUEST_FAILED',

  GET_PROFILE_REQUEST_PENDING = 'GET_PROFILE_REQUEST_PENDING',
  GET_PROFILE_REQUEST_SUCCESS = 'GET_PROFILE_REQUEST_SUCCESS',
  GET_PROFILE_REQUEST_FAILED = 'GET_PROFILE_REQUEST_FAILED',

  LOGOUT_REQUEST_SUCCESS = 'LOGOUT_REQUEST_SUCCESS',

  SET_USER_STATE = 'SET_USER_STATE'
}