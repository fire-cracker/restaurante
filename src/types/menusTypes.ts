export interface IMenu {
  id: number
  name: string
  image: string
  price: number
  recipe: string
  type: string
}

export interface IMenuState {
  menus: IMenu | null
  count: number
  fetching: boolean
  fetched: boolean
}
