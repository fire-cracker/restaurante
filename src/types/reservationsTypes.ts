export interface IReservation {
  id: number
  userId: string
  date: string
  time: string
  type: string
  price: number
  persons: number
}
export interface INewReservation {
  date: string
  time: string
  type: string
  persons: number
}

export interface IReservationState {
  reservation: IReservation | null
  fetching: boolean
  fetched: boolean
}

export interface IStripeCharge {
  id: string
  object: string
  amount: 2000
  paid: boolean
  receipt_email: string
  receipt_number: string
  receipt_url: string
  status: string
}
