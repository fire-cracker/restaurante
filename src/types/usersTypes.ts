export interface IUser {
  id: string
  username: string
  email: string
  role: 'customer' | 'admin'
}

export interface IUserState {
  user: IUser | null
  logingIn: boolean
  isLoggedIn: boolean
}