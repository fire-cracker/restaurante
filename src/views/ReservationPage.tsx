import React, { FC, useState } from 'react'
import { RouteComponentProps } from 'react-router'
import { connect } from 'react-redux'
import Jumbotron from 'react-bootstrap/Jumbotron'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Card from 'react-bootstrap/Card'
import Col from 'react-bootstrap/Col'
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'

interface IProps extends RouteComponentProps {
  history: any
  setReservation: any
}

const ReservationPage: FC<IProps> = ({ history, setReservation }) => {
  const [validated, setValidated] = useState(false)
  const [reservationState, setReservationState] = useState({
    date: '',
    time: '',
    persons: 1,
    type: 'breakfast'
  })

  const onhandleChange = ({
    target: { name, value }
  }: {
    target: { name: string; value: any }
  }) => {
    setReservationState(prevState => ({
      ...prevState,
      [name]: value
    }))
    console.log('state>>>>', reservationState)
  }

  const handleSubmit = (event: any) => {
    const form = event.currentTarget
    if (form && form.checkValidity() === false) {
      event.preventDefault()
      event.stopPropagation()
    }

    setValidated(true)
    setReservation(reservationState)
    history.push('/checkout')
  }

  return (
    <Jumbotron fluid className="p-0">
      <Container fluid>
        <Row className="top-banner"></Row>
        <Row className="reservation-wrapper px-0">
          <Container className="container-body d-flex flex-column justify-content-center align-items-center">
            <Card className="mt-n5">
              <Card.Body className="">
                <Row>
                  <Col className="form-col justify-content-center align-items-center" lg={8}>
                    <Row className="reservation-header mb-4 flex-column justify-content-center align-items-center">
                      <Col>
                        <h4 className="text-darkkhaki">FOOD AT FIRST SIGHT</h4>
                      </Col>
                      <Col>
                        <h2>Our Restaurant Menu</h2>
                      </Col>
                      <Col className="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center">
                        <p className="line my-2 mx-1">&nbsp;</p>
                        <p className="icon">
                          <img
                            src="/images/entwined.svg"
                            alt="entwined"
                            className="fa fa-cutlery"></img>
                        </p>
                        <p className="line my-2 mx-1">&nbsp;</p>
                      </Col>
                    </Row>
                    <Row className="justify-content-center align-items-center">
                      <Form noValidate validated={validated} onSubmit={handleSubmit}>
                        <Form.Row>
                          <Form.Group as={Col}>
                            <Form.Label></Form.Label>
                            <Form.Control as="select" name="persons" onChange={onhandleChange}>
                              <option>1 person</option>
                              <option>2 persons</option>
                              <option>3 persons</option>
                              <option>4 persons</option>
                              <option>5 persons</option>
                              <option>6 persons</option>
                            </Form.Control>
                          </Form.Group>
                          <Form.Group as={Col} md="4">
                            <Form.Label></Form.Label>
                            <Form.Control
                              required
                              name="date"
                              type="date"
                              placeholder="Date"
                              onChange={onhandleChange}
                            />
                            <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
                          </Form.Group>
                          <Form.Group as={Col} md="4">
                            <Form.Label></Form.Label>
                            <Form.Control
                              required
                              name="time"
                              type="time"
                              placeholder="Time"
                              onChange={onhandleChange}
                            />
                            <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
                          </Form.Group>
                        </Form.Row>
                        <Form.Row>
                          <Form.Group as={Col}>
                            <Form.Label></Form.Label>
                            <Form.Control as="select" name="type" onChange={onhandleChange}>
                              <option>Breakfast</option>
                              <option>Lunch</option>
                              <option>Dinner</option>
                              <option>Drinks</option>
                            </Form.Control>
                          </Form.Group>
                        </Form.Row>

                        <Form.Row className="justify-content-center align-items-center">
                          <Button className="bg-black" type="submit">
                            Book A Table
                          </Button>
                        </Form.Row>
                      </Form>
                    </Row>
                  </Col>
                  <Col lg={3}>
                    <Card>Hello</Card>
                  </Col>
                </Row>
              </Card.Body>
            </Card>
          </Container>
        </Row>
      </Container>
    </Jumbotron>
  )
}

const mapStateToProps = () => ({})
const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(ReservationPage)
