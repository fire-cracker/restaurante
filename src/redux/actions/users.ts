import { Dispatch } from 'redux'
import { toast } from 'react-toastify'

import axios from '../../utils/axiosConfig'
import { ActionType, Action } from '../../types/actionsTypes'
import { IUser } from '../../types/usersTypes'

const { LOGIN_REQUEST_PENDING, LOGIN_REQUEST_SUCCESS, LOGIN_REQUEST_FAILED, SIGNUP_REQUEST_PENDING, SIGNUP_REQUEST_SUCCESS, SIGNUP_REQUEST_FAILED, SET_LOGIN_STATE } = ActionType

const api = process.env.REACT_APP_API_URL
console.log('api>>>', api)

// export const setLoggedInState = (isLoggedIn, data) => ({
//   type: SET_LOGIN_STATE,
//   payload: data,
//   isLoggedIn
// });

export const loginRequestPending = () => ({
  type: LOGIN_REQUEST_PENDING,
})

export const loginRequestSuccess = (data: IUser) => ({
  type: LOGIN_REQUEST_SUCCESS,
  payload: data,
})

export const loginRequestFailed = () => ({
  type: LOGIN_REQUEST_FAILED,
})

export const signupRequestPending = () => ({
  type: SIGNUP_REQUEST_PENDING,
})

export const signupRequestSuccess = (data: IUser) => ({
  type: SIGNUP_REQUEST_SUCCESS,
  payload: data,
})

export const signupRequestFailed = () => ({
  type: SIGNUP_REQUEST_FAILED,
})

export const login = (email: string, password: string) => async (dispatch: Dispatch<Action<any>>) => {
  try {
    dispatch(loginRequestPending())
    const {
      data: {
        data: { user, token },
      },
    } = await axios.post(`${api}/auth/login`, { email, password })
    localStorage.setItem('token', token)
    dispatch(loginRequestSuccess(user))
    return user
  } catch (error) {
    dispatch(loginRequestFailed())
    toast.error(error.response.data.data.message)
    throw error
  }
}

export const signup = (username: string, email: string, password: string) => async (dispatch: Dispatch<Action<any>>) => {
  try {
    dispatch(signupRequestPending())
    const {
      data: {
        data: { user, token },
      },
    } = await axios.post(`${api}/auth/signup`, { username, email, password })
    localStorage.setItem('token', token)
    dispatch(signupRequestSuccess(user))
    return user
  } catch (error) {
    dispatch(signupRequestFailed())
    toast.error(error.response.data.data.message)
    throw error
  }
}
