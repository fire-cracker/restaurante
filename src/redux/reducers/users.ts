import { ActionType, Action } from '../../types/actionsTypes'
import { IUserState } from '../../types/usersTypes'

const {
  LOGIN_REQUEST_PENDING,
  LOGIN_REQUEST_SUCCESS,
  LOGIN_REQUEST_FAILED,
  SIGNUP_REQUEST_PENDING,
  SIGNUP_REQUEST_SUCCESS,
  SIGNUP_REQUEST_FAILED,
  SET_USER_STATE,
  LOGOUT_REQUEST_SUCCESS
} = ActionType

const initialState: IUserState = {
  user: null,
  logingIn: false,
  isLoggedIn: localStorage.token ? true : false
}
export default (state = initialState, action: Action<any>): IUserState => {
  switch (action.type) {
    case SET_USER_STATE:
      return {
        ...state,
        user: action.payload,
        isLoggedIn: true
      }

    case LOGIN_REQUEST_PENDING:
      return {
        ...state,
        logingIn: true
      }

    case LOGIN_REQUEST_SUCCESS:
      return {
        ...state,
        user: action.payload,
        isLoggedIn: true,
        logingIn: false
      }

    case LOGIN_REQUEST_FAILED:
      return {
        ...state,
        isLoggedIn: false,
        logingIn: false
      }

    case SIGNUP_REQUEST_PENDING:
      return {
        ...state,
        logingIn: true
      }

    case SIGNUP_REQUEST_SUCCESS:
      return {
        ...state,
        user: action.payload,
        isLoggedIn: true,
        logingIn: false
      }

    case SIGNUP_REQUEST_FAILED:
      return {
        ...state,
        isLoggedIn: false,
        logingIn: false
      }

    case LOGOUT_REQUEST_SUCCESS:
      return {
        ...state,
        user: null,
        isLoggedIn: false,
        logingIn: false
      }

    default:
      return state
  }
}
