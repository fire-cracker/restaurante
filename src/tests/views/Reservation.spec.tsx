import React, { Dispatch, SetStateAction } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import ReservationPage from '../../views/ReservationPage'
import { IUser, IUserState } from '../../types/usersTypes'
import { INewReservation } from '../../types/reservationsTypes'
import { user } from '../mocks/users.mock'

const createMockStore = configureMockStore([thunk])

interface IProps {
  setReservation: Dispatch<SetStateAction<INewReservation | null>>
  userState: IUserState
  onModalShow: () => void
}

describe('Login/Signup Modal', () => {
  const defaultProps: IProps = {
    setReservation: jest.fn(),
    userState: {
      user: user,
      logingIn: false,
      isLoggedIn: true
    },
    onModalShow: jest.fn()
  }

  const initialState = {
    userState: {
      user: user,
      logingIn: false,
      isLoggedIn: true
    }
  }
  const history = {
    push: jest.fn
  }

  const onhandleChange = jest.fn
  const handleSubmit = jest.fn
  const setup = (newProps?: any, newStore?: any) => {
    const props = { ...defaultProps, ...newProps }
    const store = createMockStore({
      ...initialState,
      userState: { ...initialState.userState, ...newStore }
    })

    return render(
      <BrowserRouter>
        <Provider store={store}>
          <ReservationPage
            {...props}
            history={history}
            onhandleChange={onhandleChange}
            handleSubmit={handleSubmit}
          />
        </Provider>
      </BrowserRouter>
    )
  }

  afterAll(() => {
    jest.clearAllMocks()
    cleanup()
  })

  test('renders Reservation Page', () => {
    const wrapper = setup()
    expect(wrapper).toMatchSnapshot()
    const { getByText } = wrapper
    const linkElement = getByText('BOOK YOUR TABLE')
    expect(linkElement).toBeInTheDocument()
  })

  test('user is redirected to homepage if not looged in', async () => {
    const wrapper = setup(
      { userState: { ...defaultProps.userState, isLoggedIn: false } },
      { isLoggedIn: false }
    )
    const { findByText } = wrapper
    const linkElement = findByText('BOOK YOUR TABLE')
    waitFor(() => {
      expect(linkElement).not.toBeInTheDocument()
    })
  })

  test('should trigger onchange when date is changed', async () => {
    const wrapper = setup()
    const { getByTestId } = wrapper
    const dateInput = getByTestId('date')
    expect(dateInput).toBeInTheDocument()
    await userEvent.type(dateInput, '04/04/2020')
    waitFor(() => {
      expect(dateInput).toHaveValue('04/04/2020')
      expect(onhandleChange).toHaveBeenCalledTimes(10)
    })
  })

  it('should submit reservation if input is correct', async () => {
    const wrapper = setup()
    const { getByText, getByTestId } = wrapper
    const dateInput = getByTestId('date')
    const personsInput = getByTestId('persons')
    const timeInput = getByTestId('time')
    const typeInput = getByTestId('type')
    await userEvent.type(dateInput, '04/04/2020')
    await userEvent.type(personsInput, '3 persons')
    await userEvent.type(timeInput, '3:00')
    await userEvent.type(typeInput, 'dinner')
    const reservationButton = getByText('Book A Table')
    await userEvent.click(reservationButton)
    waitFor(() => {
      expect(handleSubmit).toHaveBeenCalled()
    })
  })
})
