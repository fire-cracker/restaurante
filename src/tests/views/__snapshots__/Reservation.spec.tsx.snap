// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login/Signup Modal renders Reservation Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reservation-wrapper"
      >
        <div
          class="p-0 jumbotron jumbotron-fluid"
        >
          <div
            class="container-fluid"
          >
            <div
              class="top-banner row"
            />
            <div
              class="main-wrapper px-0 row"
            >
              <div
                class="container-body d-flex flex-column justify-content-center align-items-center container"
              >
                <div
                  class="mt-n5 card"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="form-col justify-content-center align-items-center col-lg-8"
                      >
                        <div
                          class="reservation-header mb-4 flex-column justify-content-center align-items-center row"
                        >
                          <div
                            class="col"
                          >
                            <h4
                              class="text-darkkhaki"
                            >
                              BOOK YOUR TABLE
                            </h4>
                          </div>
                          <div
                            class="col"
                          >
                            <h2>
                              Make A Reservation
                            </h2>
                          </div>
                          <div
                            class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                          >
                            <p
                              class="line my-2 mx-1"
                            >
                               
                            </p>
                            <p
                              class="icon"
                            >
                              <img
                                alt="entwined"
                                class="fa fa-cutlery"
                                src="/images/entwined.svg"
                              />
                            </p>
                            <p
                              class="line my-2 mx-1"
                            >
                               
                            </p>
                          </div>
                        </div>
                        <div
                          class="justify-content-center align-items-center row"
                        >
                          <form
                            class=""
                            novalidate=""
                          >
                            <div
                              class="form-row"
                            >
                              <div
                                class="form-group col"
                              >
                                <label
                                  class="form-label"
                                />
                                <select
                                  class="form-control"
                                  data-testid="persons"
                                  name="persons"
                                >
                                  <option>
                                    1 person
                                  </option>
                                  <option>
                                    2 persons
                                  </option>
                                  <option>
                                    3 persons
                                  </option>
                                  <option>
                                    4 persons
                                  </option>
                                  <option>
                                    5 persons
                                  </option>
                                  <option>
                                    6 persons
                                  </option>
                                </select>
                              </div>
                              <div
                                class="form-group col-md-4"
                              >
                                <label
                                  class="form-label"
                                />
                                <input
                                  class="form-control"
                                  data-testid="date"
                                  name="date"
                                  placeholder="Date"
                                  required=""
                                  type="date"
                                />
                                <div
                                  class="valid-feedback"
                                >
                                  Looks good!
                                </div>
                              </div>
                              <div
                                class="form-group col-md-4"
                              >
                                <label
                                  class="form-label"
                                />
                                <input
                                  class="form-control"
                                  data-testid="time"
                                  name="time"
                                  placeholder="Time"
                                  required=""
                                  type="time"
                                />
                                <div
                                  class="valid-feedback"
                                >
                                  Looks good!
                                </div>
                              </div>
                            </div>
                            <div
                              class="form-row"
                            >
                              <div
                                class="form-group col"
                              >
                                <label
                                  class="form-label"
                                />
                                <select
                                  class="form-control"
                                  data-testid="type"
                                  name="type"
                                >
                                  <option>
                                    Breakfast
                                  </option>
                                  <option>
                                    Lunch
                                  </option>
                                  <option>
                                    Dinner
                                  </option>
                                  <option>
                                    Drinks
                                  </option>
                                </select>
                              </div>
                            </div>
                            <div
                              class="justify-content-center align-items-center form-row"
                            >
                              <button
                                class="bg-black btn btn-primary"
                                type="submit"
                              >
                                Book A Table
                              </button>
                            </div>
                          </form>
                        </div>
                      </div>
                      <div
                        class="col-lg-3"
                      >
                        <div
                          class="card"
                        >
                          Hello
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reservation-wrapper"
    >
      <div
        class="p-0 jumbotron jumbotron-fluid"
      >
        <div
          class="container-fluid"
        >
          <div
            class="top-banner row"
          />
          <div
            class="main-wrapper px-0 row"
          >
            <div
              class="container-body d-flex flex-column justify-content-center align-items-center container"
            >
              <div
                class="mt-n5 card"
              >
                <div
                  class="card-body"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="form-col justify-content-center align-items-center col-lg-8"
                    >
                      <div
                        class="reservation-header mb-4 flex-column justify-content-center align-items-center row"
                      >
                        <div
                          class="col"
                        >
                          <h4
                            class="text-darkkhaki"
                          >
                            BOOK YOUR TABLE
                          </h4>
                        </div>
                        <div
                          class="col"
                        >
                          <h2>
                            Make A Reservation
                          </h2>
                        </div>
                        <div
                          class="divider mt-1 mb-3 text-darkkhaki d-flex flex-row justify-content-center align-items-center col"
                        >
                          <p
                            class="line my-2 mx-1"
                          >
                             
                          </p>
                          <p
                            class="icon"
                          >
                            <img
                              alt="entwined"
                              class="fa fa-cutlery"
                              src="/images/entwined.svg"
                            />
                          </p>
                          <p
                            class="line my-2 mx-1"
                          >
                             
                          </p>
                        </div>
                      </div>
                      <div
                        class="justify-content-center align-items-center row"
                      >
                        <form
                          class=""
                          novalidate=""
                        >
                          <div
                            class="form-row"
                          >
                            <div
                              class="form-group col"
                            >
                              <label
                                class="form-label"
                              />
                              <select
                                class="form-control"
                                data-testid="persons"
                                name="persons"
                              >
                                <option>
                                  1 person
                                </option>
                                <option>
                                  2 persons
                                </option>
                                <option>
                                  3 persons
                                </option>
                                <option>
                                  4 persons
                                </option>
                                <option>
                                  5 persons
                                </option>
                                <option>
                                  6 persons
                                </option>
                              </select>
                            </div>
                            <div
                              class="form-group col-md-4"
                            >
                              <label
                                class="form-label"
                              />
                              <input
                                class="form-control"
                                data-testid="date"
                                name="date"
                                placeholder="Date"
                                required=""
                                type="date"
                              />
                              <div
                                class="valid-feedback"
                              >
                                Looks good!
                              </div>
                            </div>
                            <div
                              class="form-group col-md-4"
                            >
                              <label
                                class="form-label"
                              />
                              <input
                                class="form-control"
                                data-testid="time"
                                name="time"
                                placeholder="Time"
                                required=""
                                type="time"
                              />
                              <div
                                class="valid-feedback"
                              >
                                Looks good!
                              </div>
                            </div>
                          </div>
                          <div
                            class="form-row"
                          >
                            <div
                              class="form-group col"
                            >
                              <label
                                class="form-label"
                              />
                              <select
                                class="form-control"
                                data-testid="type"
                                name="type"
                              >
                                <option>
                                  Breakfast
                                </option>
                                <option>
                                  Lunch
                                </option>
                                <option>
                                  Dinner
                                </option>
                                <option>
                                  Drinks
                                </option>
                              </select>
                            </div>
                          </div>
                          <div
                            class="justify-content-center align-items-center form-row"
                          >
                            <button
                              class="bg-black btn btn-primary"
                              type="submit"
                            >
                              Book A Table
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                    <div
                      class="col-lg-3"
                    >
                      <div
                        class="card"
                      >
                        Hello
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
